ImageFeed/Scenes/Login/OAuthCodeViewController.swift

```diff
@@ -0,0 +1,210 @@
+        setupWebView()
+    }
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+
+        startObservingProgress()
+    }
+
+    override func viewDidDisappear(_ animated: Bool) {
+        super.viewDidDisappear(animated)
+
+        stopObservingProgress()
+    }
+
+    // MARK: View components
+    private let webView = WKWebView()
+
+    private lazy var barBackButton: UIBarButtonItem = {
+        let button = UIBarButtonItem(
+           image: .asset(.backIcon).withTintColor(.asset(.ypBlack)),
+           style: .plain,
+           target: self,
+           action: #selector(backPressed))
+
+        button.tintColor = .asset(.ypBlack)
+
+        return button
+    }()
+
+    private var progressView: UIProgressView = {
+        let progressView = UIProgressView(progressViewStyle: .bar)
+
+        progressView.tintColor = .asset(.ypBlack)
+
+        return progressView
+    }()
+}
+
+// MARK: - Observe Progress
+
+extension OAuthCodeViewController {
+    private func startObservingProgress() {
+        observations.append(
+            webView.observe(\.estimatedProgress) { [weak self] _, _ in
+                self?.updateProgress()
+            }
+        )
+    }
+
+    private func stopObservingProgress() {
+        observations = []
+    }
+}
+
+// MARK: - WebView
+
+extension OAuthCodeViewController {
+    private func setupWebView() {
+        webView.navigationDelegate = self
+
+        guard var components = URLComponents(string: .k(.authorizeURL)) else {
+            fatalError("Can't construct URLComponents for authorizeURL")
+        }
+
+        components.queryItems = [
+            URLQueryItem(name: "client_id", value: .k(.accessKey)),
+            URLQueryItem(name: "redirect_uri", value: .k(.redirectURI)),
+            URLQueryItem(name: "response_type", value: "code"),
+            URLQueryItem(name: "scope", value: .k(.accessScope))
+        ]
+
+        guard let url = components.url else {
+            fatalError("Can't construct authorization URL")
+        }
+
+        let request = URLRequest(url: url)
+
+        webView.load(request)
+    }
+
+    private func updateProgress() {
+        progressView.setProgress(
+            Float(webView.estimatedProgress),
+            animated: true)
+
+        if !progressView.isHidden && webView.estimatedProgress >= 0.999 {
+            UIView.animate(withDuration: 0.4) {
+                self.progressView.alpha = 0
+            }
+        }
+
+        if progressView.isHidden && webView.estimatedProgress < 0.999 {
+            UIView.animate(withDuration: 0.4) {
+                self.progressView.alpha = 1
+            }
+        }
+    }
+}
+
+// MARK: - WKNavigationDelegate
+
+extension OAuthCodeViewController: WKNavigationDelegate {
+    func webView(
+        _ webView: WKWebView,
+        decidePolicyFor navigationAction: WKNavigationAction,
+        decisionHandler: @escaping (WKNavigationActionPolicy
+        ) -> Void) {
```

Comment:
не надо тип разбивать на две строки
```suggestion
        decisionHandler: @escaping (WKNavigationActionPolicy) -> Void
    ) {
```

Reply:
Ой, я не ту скобочку перенес :D

Reply:
Поправил!